core/domain/suggestion_services.py

```diff
@@ -1826,3 +1827,30 @@ def update_question_suggestion(
     _update_suggestion(suggestion)
 
     return suggestion
+
+
+def enqueue_contributor_ranking_notification_email_task_transactional(
```

Comment:
This isn't actually transactional without the `@transaction_services.run_in_transaction_wrapper` annotation, right? Also, I don't think it needs to be wrapped in a transaction if it's just sending an email.

Also, where will we actually call this function?

Reply:
Yeah I renamed it.

> Also, where will we actually call this function?

As I discussed wit you, this will be triggered if a submitter/reviewer achieves a new rank at the submission/review. I have not included that logic yet since it is pretty much same as the badges generation logic. So I thought of implementing that logic in the badges generation PR.